src/main/java/org/motechproject/newebodac/service/ImportService.java

```diff
@@ -0,0 +1,266 @@
+package org.motechproject.newebodac.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.text.MessageFormat;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.motechproject.newebodac.domain.CsvConfig;
+import org.motechproject.newebodac.domain.CsvField;
+import org.motechproject.newebodac.domain.KeyCommunityPerson;
+import org.motechproject.newebodac.domain.Language;
+import org.motechproject.newebodac.domain.Vaccinee;
+import org.motechproject.newebodac.domain.Visit;
+import org.motechproject.newebodac.exception.EntityNotFoundException;
+import org.motechproject.newebodac.repository.CsvConfigRepository;
+import org.motechproject.newebodac.repository.KeyCommunityPersonRepository;
+import org.motechproject.newebodac.repository.LanguageRepository;
+import org.motechproject.newebodac.repository.VaccineeRepository;
+import org.motechproject.newebodac.repository.VisitRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import org.springframework.web.multipart.MultipartFile;
+import org.supercsv.io.CsvMapReader;
+import org.supercsv.io.ICsvMapReader;
+import org.supercsv.prefs.CsvPreference;
+
+@SuppressWarnings({"PMD.CyclomaticComplexity", "PMD.CloseResource", "PMD.ExcessiveImports",
+    "PMD.AssignmentInOperand", "PMD.NcssCount"})
+@Slf4j
+@Service
+public class ImportService {
+
+  private static final String DOMAIN_PACKAGE_PREFIX = "org.motechproject.newebodac.domain.";
+
+  @Autowired
+  private CsvConfigRepository csvConfigRepository;
+
+  @Autowired
+  private VaccineeRepository vaccineeRepository;
+
+  @Autowired
+  private KeyCommunityPersonRepository keyCommunityPersonRepository;
+
+  @Autowired
+  private VisitRepository visitRepository;
+
+  @Autowired
+  private LanguageRepository languageRepository;
+
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  private final Map<Integer, String> errorMap = new HashMap<>();
```

Comment:
this should not be a class field, move it to the import method

Reply:
it was moved to keep with PMD standards. since I shortened the method overall i guess i can put it back in method